### PR TITLE
[1.10.2] Make cable collision box closer to cable shape

### PR DESCRIPTION
**Pull request by SquidDev**
1.10.2+ version of #2668 

I'm going to be honest, I haven't tested this beyond checking it compiles. My version of IntelliJ and Gradle seem to have various grievances against OpenComputers so I gave up fighting :). Any tips would be much appreciated.

### CODE REVIEW

**Comment by payonel** [comment]
i'll definitely test it for you
tips and tricks setting up OC in IDEA? Did you remember to sacrifice a goat?
I don't know the _right_ way, but it is similar to 1.7.10 in setting up the provided and compile dependencies. However, I fail to switch between builds without getting stuck. So I actually keep separate copies of the repo specifically for being able to build in IDEA and debugging.

**Comment by SquidDev** [comment]
> i'll definitely test it for you

Thanks!

> I don't know the right way, but it is similar to 1.7.10 in setting up the provided and compile dependencies.

For 1.7.10 I gave up with IDEA and just ended up doing `./gradlew runClient` (with some tweaks to ensure other mods weren't on the class path). On my machine, Gradle doesn't even compile OpenComputers - which is strange seeing Travis has no problem. Yeah, maybe time to start drawing runic circles.

**Comment by thiakil** [comment]
I've not tried the 1.7.10 version in IDEA, but the steps should be the same;

1. Run `gradlew setupDecompWorkspace` in the checked out folder (this can be done after the IDEA project is created, but requires a refresh in the gradle panel after decomp)
1. Open IDEA, select `File -> New -> Project from existing sources -> select the build.gradle file`
1. Default settings are usually fine on the popup
1. Wait for it to sync/download deps. Your `External Libraries` section should now have a bunch of libraries.
1. Run the `genIntellijRuns` task either in terminal or from the gradle panel.

**Comment by payonel** [comment]
@thiakil I don't know the nuances of IDEA and modding, but I know how I got OC working in IDEA, and I know the steps that Sangar gave me -- and it is nothing like what you said and it takes a whole lot more monkeying around with the deps to get it working. see this wiki page I made in our docs that explains it for 1.7.10: http://ocdoc.cil.li/tutorial:debug_1.7.10

**Comment by thiakil** [comment]
deleting modules and fiddling with deps is no longer needed in modern versions of IDEA at least; it's all imported from gradle.

The `idea` task will generate an IDEA project (sidenote: it's not actually part of forgegradle), but I think it is better to let your version of IDEA do it, as the gradle `idea` plugin may not be up to date with what it wants.

**Comment by payonel** [comment]
well let's not discuss that here. but thanks for your feedback

**Comment by payonel** [comment]
I was testing if 1.10 had the same jitter issue as I reported for your 1.7.10 PR, but instead it crashed :(


Note: it crashed when I placed the SECOND cable right next to the FIRST cable
```
[21:55:02] [Client thread/FATAL]: Reported exception thrown!
net.minecraft.util.ReportedException: Ticking entity
	at net.minecraft.world.World.func_72939_s(World.java:1763) ~[aid.class:?]
	at net.minecraft.client.Minecraft.func_71407_l(Minecraft.java:1789) ~[bcx.class:?]
	at net.minecraft.client.Minecraft.func_71411_J(Minecraft.java:1055) ~[bcx.class:?]
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:371) [bcx.class:?]
	at net.minecraft.client.main.Main.main(SourceFile:124) [Main.class:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_131]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_131]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_131]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_131]
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135) [launchwrapper-1.12.jar:?]
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28) [launchwrapper-1.12.jar:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_131]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_131]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_131]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_131]
	at org.multimc.onesix.OneSixLauncher.launchWithMainClass(OneSixLauncher.java:236) [NewLaunch.jar:?]
	at org.multimc.onesix.OneSixLauncher.launch(OneSixLauncher.java:297) [NewLaunch.jar:?]
	at org.multimc.EntryPoint.listen(EntryPoint.java:162) [NewLaunch.jar:?]
	at org.multimc.EntryPoint.main(EntryPoint.java:53) [NewLaunch.jar:?]
Caused by: java.lang.IllegalAccessError: tried to access method net.minecraft.block.Block.func_185492_a(Lnet/minecraft/util/math/BlockPos;Lnet/minecraft/util/math/AxisAlignedBB;Ljava/util/List;Lnet/minecraft/util/math/AxisAlignedBB;)V from class li.cil.oc.common.block.Cable$$anonfun$addCollisionBoxToList$1
	at li.cil.oc.common.block.Cable$$anonfun$addCollisionBoxToList$1.apply(Cable.scala:68) ~[Cable$$anonfun$addCollisionBoxToList$1.class:?]
	at li.cil.oc.common.block.Cable$$anonfun$addCollisionBoxToList$1.apply(Cable.scala:66) ~[Cable$$anonfun$addCollisionBoxToList$1.class:?]
	at scala.collection.IndexedSeqOptimized$class.foreach(IndexedSeqOptimized.scala:33) ~[IndexedSeqOptimized$class.class:?]
	at scala.collection.mutable.ArrayOps$ofRef.foreach(ArrayOps.scala:186) ~[ArrayOps$ofRef.class:?]
	at li.cil.oc.common.block.Cable.func_185477_a(Cable.scala:66) ~[Cable.class:?]
	at net.minecraft.block.state.BlockStateContainer$StateImplementation.func_185908_a(BlockStateContainer.java:438) ~[art$a.class:?]
	at net.minecraft.world.World.func_184144_a(World.java:1252) ~[aid.class:?]
	at net.minecraft.client.entity.EntityPlayerSP.func_189810_i(EntityPlayerSP.java:1082) ~[bnn.class:?]
	at net.minecraft.client.entity.EntityPlayerSP.func_70091_d(EntityPlayerSP.java:1003) ~[bnn.class:?]
	at net.minecraft.entity.EntityLivingBase.func_70612_e(EntityLivingBase.java:1795) ~[sf.class:?]
	at net.minecraft.entity.player.EntityPlayer.func_70612_e(EntityPlayer.java:1808) ~[zs.class:?]
	at net.minecraft.entity.EntityLivingBase.func_70636_d(EntityLivingBase.java:2217) ~[sf.class:?]
	at net.minecraft.entity.player.EntityPlayer.func_70636_d(EntityPlayer.java:512) ~[zs.class:?]
	at net.minecraft.client.entity.EntityPlayerSP.func_70636_d(EntityPlayerSP.java:959) ~[bnn.class:?]
	at net.minecraft.entity.EntityLivingBase.func_70071_h_(EntityLivingBase.java:2013) ~[sf.class:?]
	at net.minecraft.entity.player.EntityPlayer.func_70071_h_(EntityPlayer.java:233) ~[zs.class:?]
	at net.minecraft.client.entity.EntityPlayerSP.func_70071_h_(EntityPlayerSP.java:167) ~[bnn.class:?]
	at net.minecraft.world.World.func_72866_a(World.java:1967) ~[aid.class:?]
	at net.minecraft.world.World.func_72870_g(World.java:1937) ~[aid.class:?]
	at net.minecraft.world.World.func_72939_s(World.java:1750) ~[aid.class:?]
	... 18 more
```

```
java.lang.IllegalAccessError: tried to access method net.minecraft.block.Block.func_185492_a(Lnet/minecraft/util/math/BlockPos;Lnet/minecraft/util/math/AxisAlignedBB;Ljava/util/List;Lnet/minecraft/util/math/AxisAlignedBB;)V from class li.cil.oc.common.block.Cable$$anonfun$addCollisionBoxToList$1
	at li.cil.oc.common.block.Cable$$anonfun$addCollisionBoxToList$1.apply(Cable.scala:68)
	at li.cil.oc.common.block.Cable$$anonfun$addCollisionBoxToList$1.apply(Cable.scala:66)
	at scala.collection.IndexedSeqOptimized$class.foreach(IndexedSeqOptimized.scala:33)
	at scala.collection.mutable.ArrayOps$ofRef.foreach(ArrayOps.scala:186)
	at li.cil.oc.common.block.Cable.func_185477_a(Cable.scala:66)
	at net.minecraft.block.state.BlockStateContainer$StateImplementation.func_185908_a(BlockStateContainer.java:438)
	at net.minecraft.world.World.func_184144_a(World.java:1252)
	at net.minecraft.client.entity.EntityPlayerSP.func_189810_i(EntityPlayerSP.java:1082)
	at net.minecraft.client.entity.EntityPlayerSP.func_70091_d(EntityPlayerSP.java:1003)
	at net.minecraft.entity.EntityLivingBase.func_70612_e(EntityLivingBase.java:1795)
	at net.minecraft.entity.player.EntityPlayer.func_70612_e(EntityPlayer.java:1808)
	at net.minecraft.entity.EntityLivingBase.func_70636_d(EntityLivingBase.java:2217)
	at net.minecraft.entity.player.EntityPlayer.func_70636_d(EntityPlayer.java:512)
	at net.minecraft.client.entity.EntityPlayerSP.func_70636_d(EntityPlayerSP.java:959)
	at net.minecraft.entity.EntityLivingBase.func_70071_h_(EntityLivingBase.java:2013)
	at net.minecraft.entity.player.EntityPlayer.func_70071_h_(EntityPlayer.java:233)
	at net.minecraft.client.entity.EntityPlayerSP.func_70071_h_(EntityPlayerSP.java:167)
	at net.minecraft.world.World.func_72866_a(World.java:1967)
	at net.minecraft.world.World.func_72870_g(World.java:1937)
	at net.minecraft.world.World.func_72939_s(World.java:1750)
	at net.minecraft.client.Minecraft.func_71407_l(Minecraft.java:1789)
	at net.minecraft.client.Minecraft.func_71411_J(Minecraft.java:1055)
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:371)
	at net.minecraft.client.main.Main.main(SourceFile:124)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.multimc.onesix.OneSixLauncher.launchWithMainClass(OneSixLauncher.java:236)
	at org.multimc.onesix.OneSixLauncher.launch(OneSixLauncher.java:297)
	at org.multimc.EntryPoint.listen(EntryPoint.java:162)
	at org.multimc.EntryPoint.main(EntryPoint.java:53)
```

**Comment by SquidDev** [comment]
I've hypothetically fixed the jitter (like in #2668). 

With respect to the above crash: that looks very much like a Scala bug: such code should either fail to compile or generate the appropriate hidden wrappers. I'm not even sure why it's generating as many anonymous classes as it is :/. Anyway..., I'll see if using my own wrapper method is sufficient to prevent this.

Edit: Looks like an instance of scala/bug#6934.

**Comment by SquidDev** [comment]
Well, it's a little grim but that appears to work.

File: src/main/scala/li/cil/oc/common/block/Cable.scala [hunk 1]
```diff
@@ -61,13 +61,7 @@ class Cable(protected implicit val tileTag: ClassTag[tileentity.Cable]) extends
   override def getBoundingBox(state: IBlockState, world: IBlockAccess, pos: BlockPos): AxisAlignedBB = Cable.bounds(world, pos)
 
   override def addCollisionBoxToList(state : IBlockState, world : World, pos : BlockPos, bigBox : AxisAlignedBB, boxes : util.List[AxisAlignedBB], entity : Entity) = {
```
Comment (thiakil) [inline src/main/scala/li/cil/oc/common/block/Cable.scala]: `bigBox` should probably be renamed to match the MCP name of `entityBox`, for clarity

**Comment by thiakil** [comment]
You could AT the `addCollisionBoxToList` function to make it public if you prefer using it

**Comment by payonel** [comment]
merged via 9c00fcfc285465e180792eef44c02f94dcc4a256

**Pull request closed without merging**